cnd/migrations/2020-05-06-105000_created-swap-tables/up.sql

```diff
@@ -0,0 +1,64 @@
+-- Your SQL goes here
+
+CREATE TABLE swaps
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    local_swap_id UNIQUE	NOT NULL,
+    role			NOT NULL,
+    counterparty_peer_id	NOT NULL
+);
+
+CREATE TABLE secret_hashes
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    swap_id INTEGER		NOT NULL,
+    secret_hash		NOT NULL,
+    FOREIGN KEY (swap_id)	REFERENCES swaps(id)
+);
+
+CREATE TABLE shared_swap_ids
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    swap_id INTEGER		NOT NULL,
+    shared_swap_id		NOT NULL,
+    FOREIGN KEY (swap_id)	REFERENCES swaps(id)
+);
+
+CREATE TABLE address_hints
```

Comment:
We could generalize this table to an `address_book` where we store addresses or peer IDs.

It is really only a "hint" for the initial request.
Anytime we receive a request, we could potentially load all addresses for this peer from this table and try all of them.

For now, we could just name it `address_book` but not anything besides saving entries.

Reply:
That is a valid solution, there would not be a relation to the swap then, though (we would just imply the relationship through the peer-id). 
I actually tried to model a relationship from `swap` to `address_hint` on the `counterparty_peer_id` but diesel did not let me do that.

If we are OK with just having a separate table `address_book` where we save `peer_id` `address_hint` relationships we can do so. 
We can also relate entries in there to a swap on the `swap_id`, but from an `address_book` point of view this relation does not make sense.

Reply:
For simplicity reasons I would actually keep it as currently modelled. Or is the `address_book` relevant for recovery?

Reply:
I don't think there need to be any relations. We are just storing addresses of peers :)